bootstrap/lib/logger.py

```diff
@@ -248,23 +253,37 @@ def log_dict(self, group, dictionary, description='', stack_displacement=2, shou
             return -1
 
         flat_dictionary = self._flatten_dict(dictionary)
-        if self._check_table_exists(group):
+        if self._check_table_exists(group).fetchone():
             columns = self._list_columns(group)
-            for key in flat_dictionary.keys():
+            for key in flat_dictionary:
                 if key not in columns:
-                    self.log_message('Key "{}" is unknown. New keys are not allowed'.format(key), log_level=self.ERROR)
+                    self.log_message(f'Key "{key}" is unknown. New keys are not allowed', log_level=self.ERROR)
             for column_name in columns:
                 if column_name not in flat_dictionary:
-                    self.log_message('Key "{}" not in the dictionary to be logged'.format(key), log_level=self.ERROR)
+                    self.log_message(f'Key "{column_name}" not in the dictionary to be logged', log_level=self.ERROR)
         else:
             self._create_table(group)
-            for key, value in flat_dictionary.keys():
+            for key, value in flat_dictionary.items():
                 self._add_column(group, key, value)
 
         self._insert_row(group, flat_dictionary)
+
         if should_print:
             self.log_dict_message(group, dictionary, description, stack_displacement + 1, log_level)
 
+    def select(self, group, columns=None):
+        table_name = self._get_internal_table_name(group)
+        table_columns = self._list_columns(group)
+        if columns is None:
+            column_string = '*'
+        else:
+            for c in columns:
+                if c not in table_columns:
+                    self.log_message(f'Unknown column "{c}"', log_level=self.ERROR)
+            column_string = ', '.join([f'"{c}"' for c in columns])
```

Comment:
Also not SQL safe. But I guess we'll have to accept columns and table names are passive to injection...

Reply:
yes, you can't use parameters substitution in sqlite for table and columns names in sqlite, only for values

Reply:
maybe we could maintain a state with the current valid tables in the database...